doc/pmon/sonic_thermal_control_test_plan.md

```diff
@@ -0,0 +1,250 @@
+# Thermal Control Test Plan
+
+## Table of Content
+
+- [Overview](#overview)
+- [Scope](#scope)
+- [Test Structure](#test-structure)
+  - [Setup Configuration](#setup-configuration)
+  - [Ansible and Pytest](#ansible-and-pytest)
+- [Test Cases](#test-cases)
+  - [Show FAN Status Test](#show-fan-status-test)
+  - [Show Thermal Status Test](#show-thermal-status-test)
+  - [FAN Test](#fan-test)
+  - [PSU Absence Test](#psu-absence-test)
+  - [Invalid Policy Format Load Test](#invalid-policy-format-load-test)
+  - [Invalid Policy Value Load Test](#invalid-policy-value-load-test)
+
+
+## Overview
+
+The purpose is to test functionality of thermal control feature on the SONiC switch DUT. The thermal control feature contains 3 major functions: FAN status monitor, thermal status monitor and thermal control policy management.
+
+- FAN status monitor read FAN status via platform API every 60 seconds and save it to redis database. User can fetch FAN status via command line `show platform fanstatus`.
+- Thermal status monitor read thermal status via platform API every 60 seconds and save it to redis database. User can fetch thermal status via command line `show platform temperature`.
+- The thermal control policy is defined in a JSON file and loaded by thermal control daemon in pmon docker. Thermal control daemon collects thermal information and matches thermal conditions. Once some thermal conditions match, related thermal actions will be triggered. 
+
+More detail on thermal control feature can be found in this [document](https://github.com/keboliu/SONiC/blob/thermal_control_design/thermal-control-design.md).
+
+## Scope
+
+This test is targeting a running SONiC system with fully functioning configuration. The purpose of the test is functional testing of thermal control on SONiC system, making sure that FAN status and thermal status can be shown to user and correct actions are executed once predefined thermal policy conditions match.
+
```

Comment:
Do you plan to verify different JSON files for different 'thermal policy'? if not, i suggest you will use the one defined in the system and some other functionality which can expose more flows to be validated.

Reply:
No, I am not going to verify different JSON files for different 'thermal policy'. The JSON files used by test is almost the same as the one defined in the system, except that it adds some actions to help verify the test result.